web/src/app/chat/components/input/ActionManagement.tsx

```diff
@@ -177,6 +185,36 @@ function ActionItem({
               </span>
             </div>
             <div className="flex items-center gap-2">
+              {/* OAuth Authentication Indicator */}
+              {tool?.oauth_config_id && oauthStatus && (
+                <div
+                  className="flex items-center gap-2 transition-opacity duration-200 opacity-0 group-hover:opacity-100"
+                  onClick={(e) => {
+                    e.stopPropagation();
+                    if (!oauthStatus.has_token || oauthStatus.is_expired) {
+                      onOAuthAuthenticate?.();
+                    }
+                  }}
+                >
+                  {!oauthStatus.has_token || oauthStatus.is_expired ? (
+                    <FiKey
+                      size={16}
+                      className="transition-colors cursor-pointer text-yellow-500 hover:text-yellow-600"
+                      title={
+                        oauthStatus.is_expired
+                          ? "OAuth token expired - click to re-authenticate"
+                          : "Click to authenticate"
+                      }
+                    />
```

Comment:
@duo-onyx's created a new key icon in Figma!

If you want to import it, you can either import it via the `web/src/icons/README.md` instructions (manual and a bit tiresome), OR you could use the Figma MCP tool to import it.

P.s., if you do the latter, you should remove the `stroke="..."` and `strokeOpacity="..."` props from the imported `key.tsx` file. Those props hardcode the colours and opacities into the component, thus preventing the end-developer from setting the colour of the icon themselves.

Reply:
We should ideally move away from using icons provided by `lucide-icons`, `react-icons`, etc. in our codebase, especially as we start bolting down our design language and aesthetic.